src/CodeGen_Xtensa.cpp

```diff
@@ -1752,9 +1752,11 @@ HALIDE_ALWAYS_INLINE uint8x64_t halide_xtensa_sat_narrow_with_rounding_shift_u8(
 HALIDE_ALWAYS_INLINE int16x32_t halide_xtensa_narrow_with_rounding_shift_i16(const int32x32_t& a, uint32_t shift) {
   xb_vecNx48 wide = convert_to_int48x32_t_from_int32x32_t(a);
   // Add rounding factor.
-  int32_t half_shift_1 = (shift - 1) >> 1;
-  int32_t half_shift_2 = (shift - 1) - half_shift_1;
-  IVP_MULANX16(wide, int16x32_t(1 << half_shift_1), int16x32_t(1 << half_shift_2));
+  const uint16_t half_shift_1 = (shift - 1) >> 1;
+  const uint16_t half_shift_2 = (shift - 1) - half_shift_1;
```

Comment:
I think there should be a comment explaining why we split this into two different constants.